isisd/isisd.c

```diff
@@ -493,100 +656,159 @@ int area_clear_net_title(struct vty *vty, const char *net_title)
  * 'show isis interface' command
  */
 
-int show_isis_interface_common(struct vty *vty, const char *ifname, char detail)
+int show_isis_interface_common(struct vty *vty, const char *ifname, char detail,
+			       char *vrf_name, bool all_vrf)
 {
-	struct listnode *anode, *cnode;
+	struct listnode *anode, *cnode, *mnode, *inode;
 	struct isis_area *area;
 	struct isis_circuit *circuit;
+	struct isis *isis = NULL;
 
-	if (!isis) {
+	if (!im) {
 		vty_out(vty, "IS-IS Routing Process not enabled\n");
 		return CMD_SUCCESS;
 	}
+	if (vrf_name) {
```

Comment:
Let's find a way to not cut-n-paste the same code for the case of vrf default -vs- a specified single vrf

Reply:
Sure, will find a way and do it for the case of a single vrf -vs- the default vrf.